ui/app/src/App.tsx

```diff
@@ -11,48 +11,38 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { DashboardResource } from '@perses-ui/core';
-import { Box, Theme } from '@mui/material';
-import { SxProps } from '@mui/system/styleFunctionSx/styleFunctionSx';
-import { PluginRegistry } from './context/plugin-registry';
-import ViewDashboard from './views/dashboard/ViewDashboard';
-import AlertErrorFallback from './components/AlertErrorFallback';
+import { Box } from '@mui/material';
+import { ErrorAlert } from '@perses-dev/components';
+import { PluginRegistry, PluginBoundary } from '@perses-dev/plugin-system';
+import ViewDashboard from './views/ViewDashboard';
 import { DataSourceRegistry } from './context/DataSourceRegistry';
-import { useSampleData } from './utils/temp-sample-data';
 import Header from './components/Header';
-import { pluginRuntime } from './model/plugin-runtime';
-
-const appStyle: SxProps<Theme> = {
-  display: 'flex',
-  flexDirection: 'column',
-  height: '100vh',
-};
+import { useBundledPlugins } from './model/bundled-plugins';
 
 function App() {
-  const dashboard = useSampleData<DashboardResource>(
-    new URLSearchParams(window.location.search).get('dashboard') || 'node-exporter-full'
-  );
-  if (dashboard === undefined) {
-    return null;
-  }
+  const { getInstalledPlugins, importPluginModule } = useBundledPlugins();
 
   return (
-    <Box sx={appStyle}>
+    <Box
+      sx={{
+        display: 'flex',
+        flexDirection: 'column',
+        height: '100vh',
+      }}
+    >
       <Header />
       <Box
         sx={{
           flexGrow: 1,
           overflow: 'hidden',
         }}
       >
-        <PluginRegistry
-          loadingFallback="Loading..."
-          ErrorFallbackComponent={AlertErrorFallback}
-          runtime={pluginRuntime}
-        >
-          <DataSourceRegistry>
-            <ViewDashboard resource={dashboard} />
-          </DataSourceRegistry>
+        <PluginRegistry getInstalledPlugins={getInstalledPlugins} importPluginModule={importPluginModule}>
```

Comment:
Yeah, I'm not sure about Datasources yet to be honest, but the `PluginRegistry` here is because I was thinking of "what plugins are available and loaded" as a piece of global state that's shared across the application. Basically we set it up here once and then all the package underneath it can use plugins. Datasources are here for the same reason (for now) but maybe we'll find out that they're feature-specfic. I feel like I need to think a bit more on what a "Datasource" actually is/should be and start a discussion.